test/box-luatest/gh_4693_formats_for_standalone_tuples_test.lua

```diff
@@ -37,6 +37,21 @@ g.test_box_tuple_format_new = function()
     end)
 end
 
+-- Checks that improper use of `box.internal.tuple_format.new` doesn't crash.
+g.test_box_internal_tuple_format_new = function(cg)
+    cg.server:exec(function()
+        t.assert_error_msg_equals(
+            "expected table as 1 argument",
+            box.internal.tuple_format.new)
+        t.assert_error_msg_equals(
+            "expected table as 1 argument",
+            box.internal.tuple_format.new, 1984)
+        t.assert_error_msg_equals(
+            "Wrong space format",
+            box.internal.tuple_format.new, {type = 'test'})
+    end)
+end
+
```

Comment:
It still can be easily crashed. 

```lua
tarantool> box.internal.tuple_format.new({constraint = 1})
tarantool: ./src/box/field_def.c:584: field_def_array_decode: Assertion `mp_typeof(**data) == MP_ARRAY' failed.
Aborted (core dumped)
```

I thought we can hide it like this:

```diff
@@ -2,12 +2,15 @@ local utils = require('internal.utils')

 local internal = box.internal

+local tuple_format_new = box.internal.tuple_format.new
+box.internal.tuple_format.new = nil
+
 -- new() needs a wrapper in Lua, because format normalization needs to be done
 -- in Lua.
 box.tuple.format.new = function(format)
     utils.check_param(format, 'format', 'table')
     format = box.internal.space.normalize_format(nil, nil, format, 2)
-    return box.internal.tuple_format.new(format)
+    return tuple_format_new(format)
 end
```

But unfortunately this function is used from other Lua modules/tests. Also I guess we have a lot of issues of this kind with other internal functions. If the issue we can easily can get this function in auto-completion - may be we'd better to disable auto-completion for `box.internal`. Let's ask also @locker opinion on the topic.